src/client/datascience/types.ts

```diff
@@ -272,6 +273,7 @@ export const INotebookExecutionLogger = Symbol('INotebookExecutionLogger');
 export interface INotebookExecutionLogger extends IDisposable {
     preExecute(cell: ICell, silent: boolean): Promise<void>;
     postExecute(cell: ICell, silent: boolean, language: string, resource: Uri): Promise<void>;
+    nativePostExecute(cell: NotebookCell): Promise<void>;
```

Comment:
Can we make this an optional method, this way you don't need to modify all other classes.


```suggestion
    nativePostExecute?(cell: NotebookCell): Promise<void>;
```